_applications/my-first-radanalytics-app.adoc

```diff
@@ -0,0 +1,113 @@
+
+It is worth mentioning that in a normal environment you would not want to run extended calculations during an HTTP request unless they can be completed quickly. For the purposes of this tutorial however, it simplifies the implementation details. As you build further applications from the material provided here please keep this simplification of design in mind.
+
+Regardless of the implementation language or the HTTP framework, all of the tutorial applications linked from this page have the same general architecture. This diagram shows the major components involved with deployment and usage of the microservice:
+
+pass:[<img src="/assets/my-first-radanalytics-app/sparkpi-architecture.svg" class="img-responsive center-block">]
+
+**Component key**
+
+1. Computer with a web browser or other method for making HTTP requests
+
+2. The "SparkPi" microservice
+
+3. An Apache Spark cluster
+
+Although this is three different components, the main one that you will be concerned with throughout the tutorial is #2. You should already have a computer with the necessary tools installed for making Pi requests, editing your source files and commanding OpenShift to deploy your microservice. The Apache Spark cluster will be created automatically for you by the http://github.com/radanalyticsio/oshinko-s2i[Oshinko source-to-image tools].
+
+As https://docs.openshift.org/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[source-to-image] technology is fundamental to the operation of this tutorial, let's take a closer look at its details. Source-to-image provides a convenient way for you to produce container images directly from your source repositories. This work is handled by language specific builder images which can ingest your source code, assemble it and then produce an application image which is ready to be deployed. In the case of the Oshinko source-to-image builders, an Apache Spark cluster will also be created dynamically for your application when it runs.
+
+[[setup]]
+== Setup your coding environment
+
+To utilize the source-to-image workflow you will need to have a https://git-scm.com[Git] repository to store your work. This repository is then used by the source-to-image builders to create the final application image which will be run on OpenShift. For the purposes of this tutorial your repository will need to be accessible by the OpenShift cluster, the easiest way to do this is by using an external service like https://github.com[GitHub] or https://bitbucket.com[BitBucket].
```

Comment:
same kind of nit with the coma after "OpenShift cluster".   Maybe a dash as another alternative?

Totally being picky, the content is awesome!

Reply:
i just opted for a second sentence. i think that works and is probably more proper.